core/src/main/resources/jenkins/model/Jenkins/login.jelly

```diff
@@ -23,58 +23,103 @@ THE SOFTWARE.
 -->
 
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form" xmlns:x="jelly:xml" xmlns:st="jelly:stapler">
   <j:choose>
-  <j:when test="${it.setupWizard.isUsingSecurityToken()}">
-    <st:include it="${it.setupWizard}" page="authenticate-security-token"/>
-  </j:when>
-  <j:otherwise>
-  <l:layout norefresh="true">
-    <l:hasPermission permission="${app.READ}">
-      <st:include page="sidepanel.jelly" />
-    </l:hasPermission>
-    <l:main-panel>
-      <div style="margin: 2em;">
-        <!-- login form -->
-        <form name="login" action="${it.securityRealm.authenticationGatewayUrl}" method="post" style="text-size:smaller">
-          <table>
-            <tr>
-              <td>${%User}:</td>
-              <td><input type="text" name="j_username" id="j_username" autocorrect="off" autocapitalize="off" /></td>
-            </tr>
-            <tr>
-              <td>${%Password}:</td>
-              <td><input type="password" name="j_password" /></td>
-            </tr>
-            <j:if test="${app.security.name()=='SECURED' and !app.disableRememberMe}">
-              <!-- remember me service only available for Acegi -->
-              <tr>
-                <td align="right"><input id="remember_me" type="checkbox" name="remember_me" /></td>
-                <td><label for="remember_me">${%Remember me on this computer}</label></td>
-              </tr>
-            </j:if>
-          </table>
-          <input type="hidden" name="from" value="${request.getParameter('from')}" />
-          <f:submit value="${%login}" />
-          <script>
-            $('j_username').focus();
-          </script>
-        </form>
+    <j:when test="${request.getParameter('legacy')=='true'}">
+      <st:include it="${it}" page="login.legacy.jelly"/>
+    </j:when>
+    <j:otherwise>
+      <st:setHeader name="Expires" value="0" />
+      <st:setHeader name="Cache-Control" value="no-cache,no-store,must-revalidate" />
+      <st:setHeader name="X-Hudson-Theme" value="default" />
```

Comment:
What is the purpose of this header?

Reply:
I have to admit I am not sure. It had been there before so I did not removed it. However I may want to change the value to `fresh`

Reply:
https://github.com/jenkinsci-cert/jenkins/commit/7bee9394389de54ed402ddf0ccf1df14dbbeda8a is the only reference I could find. No related issues from around that time turned up. We could probably just remove this header in a separate PR and see what happens.